Source/Reporters/DataTableReporter.jsl

```diff
@@ -0,0 +1,139 @@
+			New Column("Label 4", Character, "Nominal", Formula(If(Is Missing(:Label 5) & N Items(:label separated[Row()]) >= 4, :label separated[Row()][4], N Items(:label separated[Row()]) >= 5, :label separated[Row()][4], .))),
+			New Column("Label 5", Numeric, "Nominal", Format("Best", 9), Formula(If(N Items(:label separated[Row()]) < 3, ., Regex(:label separated[Row()][N Items(:label separated[Row()])], " *([0-9]+)", "1") == "1", Num(Regex(:label separated[Row()][N Items(:label separated[Row()])], " *([0-9]+)", "\1")), .))),
+			New Column( "test expr", Expression, "None", Set Values({}) ),
+			New Column( "description", Character, "Nominal", Set Values({}) ),
+			New Column( "lre", Expression, "None", Set Values({}) ),
+			New Column( "payload", Expression, "None", Set Values({}) ),
+			New Column( "result", Character, "Nominal", Set Values({}) )
+		);
+		1;
+	);
+
+	// Function: Table
+	// Returns a reference to the results data table
+	Table = function = Method( {}, return( dt ) );
+	
+	// Function: add expression failure
+	add expression failure = Method( {label, payload=Empty()},
+		Eval( Eval Expr( dt << Add Rows( { 
+			DateTime = today(),
+			label= label,
+			test expr = "is and expression",
+			description = "was not an expression",
+			lre = Empty(),
+			payload = Expr( Expr( Name Expr( payload ) ) ),
+			result = "expression failure"
+		} ) ) );
+	);
+
+	// Function: add failure
+	add failure = Method( { label, test expr, description, mismatch, lre, payload=Empty() },	
+		Eval( Eval Expr( rw = { 
+			DateTime = today(),
+			label= label,
+			test expr = Expr( Expr( Name Expr( test expr ) ) ),
+			description = description,
+			lre = Expr( Expr( Name Expr( l ) ) ),
+			payload = Expr( Expr( Name Expr( payload ) ) ),
+			result = "failure"
+		} ) );
+		show(rw);
+		dt << Add Rows( rw );
+		wait(0);
+	);
+	
+	// Function: add unexpected throw
+	add unexpected throw = Method( {label, test expr, description, exception, payload=Empty()},
+		//show(Name Expr( text expr ));
+		Eval( Eval expr( rw = { 
+			DateTime = Expr( today() ),
+			label= label,
+			test expr = Expr( Expr( Name Expr( test expr ) ) ),
+			description = "Threw \!"" || exception || "\!"",
+			lre = Empty(),
+			payload = Expr( Expr( Name Expr( payload ) ) ),
+			result = "unexpected throw"
+		} ) );
+		show(rw);
+		dt << Add Rows( rw );
+	);
+	
+	// Function: add success
+	add success = Method( {label, test expr, description, payload=Empty()},
+		Eval( Eval Expr( dt << Add Rows( { 
+			DateTime = today(),
+			label= label,
+			test expr = Expr( Expr( Name Expr( test expr ) ) ),
+			description = description,
+			lre = Empty(),
+			payload = Expr( Expr( Name Expr( payload ) ) ),
+			result = "success"
+		} ) ) );
+	);
+	
+	// Function: N successes
+	N successes = Method( {},
+		x = Associative Array(this:dt:result, this:dt:result)<<Get Values;
+		nrows(loc(x, "success") );
```

Comment:
I think a simpler way of doing this would be to use `Get Rows Where`. Same can be applied to other `N xxx` methods.
```
N Row(this:dt << Get Rows Where(:result == "success"))
```